### PR TITLE
Feature/view org details for aatf

### PR DESCRIPTION
**Pull request by SamGoodeySFW**

### CODE REVIEW

File: src/EA.Weee.RequestHandlers/Admin/GetAatfs/GetAatfsByOperatorIdHandler.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            this.dataAccess = dataAccess;
+            this.aatfmap = map;
+        }
+        public async Task<List<AatfDataList>> HandleAsync(GetAatfsByOperatorId message)
+        {
+            authorization.EnsureCanAccessInternalArea();
+
+            List<Aatf> aatfs = await dataAccess.GetAatfs();
+
+            return aatfs
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/Admin/GetAatfs/GetAatfsByOperatorIdHandler.cs]: Add unit tests to ensure the mapper is called and that the return list contains the mapped items
And that the order of the items is correct

File: src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs [hunk 2]
```diff
@@ -46,8 +49,18 @@ public async Task<ActionResult> Details(Guid id)
             {
                 AatfData aatf = await client.SendAsync(User.GetAccessToken(), new GetAatfById(id));
                 AatfContactData contactData = await client.SendAsync(User.GetAccessToken(), new GetAatfContact(id));
+                OrganisationData organisationData = await client.SendAsync(User.GetAccessToken(), new GetOrganisationInfo(aatf.Operator.OrganisationId));
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs]: Is it possible to merge these API calls into a single API call, i.e. the GetAAtfById as that should be able to return the related entities, contact and address data. The the AatfData object can be expanded to have ContactData, Address Data, and List of related entities

File: src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs [hunk 3]
```diff
@@ -135,6 +135,7 @@ private AatfData CreateAatfData(AddAatfViewModel viewModel)
                 Guid.NewGuid(),
                 viewModel.AatfName,
                 viewModel.ApprovalNumber,
+                null,
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs]: It is possible to remove the operator parameter completely from the constructor? not sure its needed anymore

File: src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs [hunk 4]
```diff
@@ -46,8 +49,18 @@ public async Task<ActionResult> Details(Guid id)
...
 
-                AatfDetailsViewModel viewModel = mapper.Map<AatfDetailsViewModel>(new AatfDataToAatfDetailsViewModelMapTransfer(aatf, contactData));
+                List<AatfDataList> associatedAatfs = await client.SendAsync(User.GetAccessToken(), new GetAatfsByOperatorId(aatf.Operator.Id));
+
+                List<Core.Scheme.SchemeData> associatedSchemes = await client.SendAsync(User.GetAccessToken(), new GetSchemesByOrganisationId(aatf.Operator.OrganisationId));
+
+                AatfDetailsViewModel viewModel = mapper.Map<AatfDetailsViewModel>(new AatfDataToAatfDetailsViewModelMapTransfer(aatf, contactData, organisationData)
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs]: Create unit tests to check that the mapper is called with the correct parameters and that the returned view model is the model returned from the mapper

File: src/EA.Weee.Web/Areas/Admin/ViewModels/Aatf/AatfDetailsViewModel.cs [hunk 5]
```diff
@@ -19,10 +22,21 @@ public class AatfDetailsViewModel
 
         public AatfSize Size { get; set; }
 
+        public OrganisationData Organisation { get; set; }
+        
+        [AllowHtml]
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Admin/ViewModels/Aatf/AatfDetailsViewModel.cs]: Write unit test to ensure that this property has this attribute

File: src/EA.Weee.Web/Areas/Admin/Views/Aatf/Details.cshtml [hunk 6]
```diff
@@ -203,6 +205,67 @@
...
+            @foreach (var aatf in Model.AssociatedAatfs)
+            {
+                <tr class="govuk-table__row">
+                    <th class="govuk-table__header" scope="col" style="border-bottom: 0px"></th>
+                    <td class="govuk-table__cell" style="border-bottom: 0px; padding: 5px 0px 0px 0px">
+                        @Html.ActionLink(aatf.Name, "Details", "Aatf", new { id = aatf.Id }, new { })
+                    </td>
+                </tr>
+            }
+            @if (Model.AssociatedAEs.Count != 0)
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Admin/Views/Aatf/Details.cshtml]: use .Any() looks nicer :)

Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Admin/Views/Aatf/Details.cshtml]: use .Any() looks nicer :)

File: src/EA.Weee.Web/Areas/Admin/Views/Aatf/Details.cshtml [hunk 7]
```diff
@@ -203,6 +205,67 @@
...
+                    <th class="govuk-table__header" scope="col" style="border-bottom: 0px"></th>
+                    <td class="govuk-table__cell" style="border-bottom: 0px; padding: 30px 0px 0px 0px">
+                        PCS
+                    </td>
+                </tr>
+            }
+            @foreach (var scheme in Model.AssociatedSchemes)
+            {
+                <tr class="govuk-table__row">
+                    <th class="govuk-table__header" scope="col" style="border-bottom: 5px"></th>
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Admin/Views/Aatf/Details.cshtml]: Any of these common styles that we can add to the SASS sheet?

File: src/EA.Weee.Web/Areas/Admin/Mappings/ToViewModel/AatfDataToAatfDetailsViewModelMap.cs [hunk 8]
```diff
@@ -27,9 +29,24 @@ public AatfDetailsViewModel Map(AatfDataToAatfDetailsViewModelMapTransfer source
...
 
+            if (source.AssociatedAatfs != null)
+            {
+                List<AatfDataList> associatedAEs = source.AssociatedAatfs.Where(a => a.FacilityType == FacilityType.Ae).ToList();
+                source.AssociatedAatfs = source.AssociatedAatfs.Where(a => a.Id != source.AatfData.Id && a.FacilityType == FacilityType.Aatf).ToList();
+                viewModel.AssociatedAatfs = source.AssociatedAatfs;
+                viewModel.AssociatedAEs = associatedAEs;
+            }
+
+            if (source.AssociatedSchemes != null)
```
Comment (JB12298) [inline src/EA.Weee.Web/Areas/Admin/Mappings/ToViewModel/AatfDataToAatfDetailsViewModelMap.cs]: Add a tests for the null check
